packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -374,12 +376,18 @@ export class GarbageCollector implements IGarbageCollector {
 			return;
 		}
 
-		// Initialize the deleted nodes from the snapshot. This is done irrespective of whether sweep is enabled or not
+		// Initialize the deleted nodes from the snapshot. This is done irrespective of whether GC / sweep is enabled
 		// to identify deleted nodes' usage.
 		if (baseSnapshotData.deletedNodes !== undefined) {
 			this.deletedNodes = new Set(baseSnapshotData.deletedNodes);
 		}
 
+		// If GC is disabled, the GC and tombstone state shouldn't be tracked because there shouldn't be any logs or
+		// errors from GC. The exception is deleted node usage which is mentioned above.
+		if (!this.shouldRunGC) {
+			return;
```

Comment:
I think we want an early return in `initializeOrUpdateGCState` too, to avoid logging on every "connected" event (and it will say `initialized: false` every time too)